pkg/driver/driver_azure.go

```diff
@@ -502,48 +504,56 @@ func (d *AzureDriver) createVMNicDisk() (*compute.VirtualMachine, error) {
 	}
 	onARMAPISuccess(prometheusServiceSubnet, "subnet.Get")
 
-	/*
-		NIC creation
-	*/
+	NIC, err := clients.nic.Get(ctx, resourceGroupName, nicName, "")
```

Comment:
Hey Prashanth,

If this `Get()` call returns an error apart from 404 Not Found, it will result in nil pointer error at line 634 `VMParameters := d.getVMParameters(vmName, vmImageRef, *NIC.ID)` as it will have no reference to NIC.

Found this issue while I was incorporating this in OOT and wrote a UT for the same. Please let me know your thoughts about it.

Reply:
Nice catch. This is why we need tests. :) 

I have tried to fix this scenario with [this commit](https://github.com/gardener/machine-controller-manager/pull/594/commits/7c9b0d9ad6f4f56b0208381404def236bc383e9d). Please let me know if it makes sense.

Reply:
Perfect! This works like a charm!